### PR TITLE
Add "Composer Lint" job to CI pipeline

### PR DESCRIPTION
**Pull request by phansys**
Follows https://github.com/doctrine/DoctrineBundle/pull/1540#issuecomment-1161757390.

I was not able to leverage the [Coding Standards workflow]( https://github.com/phansys/doctrine.github/blob/9740cba8d0ce280768b7e0fa56dc65b143850531/.github/workflows/coding-standards.yml#L2) as I think (currently) [there is no simple way to filter a single job](https://github.community/t/path-filtering-for-jobs-and-steps/16447).

### CODE REVIEW

File: .github/workflows/composer-lint.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          php-version: "${{ matrix.php-version }}"
+          tools: composer:v2, composer-normalize:2
+        env:
+          COMPOSER_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+
+      - name: "Composer normmalize"
+        run: "composer-normalize --dry-run"
+
+      - name: "Composer validate"
+        run: "composer validate"
```
Comment (derrabus) [inline .github/workflows/composer-lint.yml:42]: Doesn't composer-normalize already run this validation step?

File: .github/workflows/composer-lint.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+
+name: "Composer Lint"
+
+on:
+  workflow_call:
+    inputs:
+      php-version:
+        description: "The PHP version to use when running the job"
+        default: "8.0"
```
Comment (derrabus) [inline .github/workflows/composer-lint.yml:9]: Why not 8.1?

```suggestion
        default: "8.1"
```

File: .github/workflows/composer-lint.yml (hunk 1 above)
Comment (phansys) [reply .github/workflows/composer-lint.yml]: Even if that is not explicitly stated in [the docs](https://github.com/ergebnis/composer-normalize#details), I guess the validation is already covered (ergebnis/composer-normalize#297, ergebnis/composer-normalize#406).

**Review by derrabus — APPROVED** [review]

**Comment by greg0ire** [comment]
Can you please test this before we merge it? Here is how you can do so: https://github.com/doctrine/DoctrineMigrationsBundle/commit/9321c8c25e53ed195e5a4a9ae30fed115a972764

**Comment by phansys** [comment]
>Can you please test this before we merge it? Here is how you can do so: https://github.com/doctrine/DoctrineMigrationsBundle/commit/9321c8c25e53ed195e5a4a9ae30fed115a972764

Thank you @greg0ire!
[Testing the build](https://github.com/doctrine/DoctrineBundle/actions/runs/2541286498) I figured out that the `paths` option is only available for these events: `push`, `pull_request`, `pull_request_target`.
What do you suggest?

**Comment by greg0ire** [comment]
Well I'm confused, since this is a pull request, isn't it?

**Review by greg0ire — CHANGES_REQUESTED** [review]
Please add a workflow template and a properties file in the `workflow-templates` directory

**Review by greg0ire — APPROVED** [review]

**Pull request merged**